cartographer/mapping/trajectory_node.h

```diff
@@ -54,6 +54,7 @@ struct TrajectoryNode {
   // node is being trimmed, it must survive until all use finishes.
   std::shared_ptr<const Data> constant_data;
 
+  const transform::Rigid3d& initial_pose;
```

Comment:
This does not belong here. It does not take part in the optimization, so it should be part of the `constant_data`. Similar for the proto. Also: Do not keep a dangling reference, this does not work.

Reply:
Oops, the reference was a typo. The initial pose _is_ used in the optimization problem: https://github.com/googlecartographer/cartographer/blob/master/cartographer/mapping_2d/sparse_pose_graph/optimization_problem.cc#L239

Reply:
Used yes, but not as an optimization variable: It is similar to e.g. `gravity_alignment`. It is taken into account but not allowed to change.